File: src/app/model/job-procedure-task.ts
import { TaskStatus } from "./task-status";
import { UserComment } from "./user-comment";
import { TaskType } from "./task-type";

export class JobProcedureTask {
  public taskType: TaskType;
  public taskName: string = '';
  public taskStatus: TaskStatus;
  public comments: UserComment[] = [];

  constructor(taskType: TaskType, taskName: string, taskStatus: TaskStatus, comments: UserComment[]) {
    this.taskType = taskType;
    this.taskName = taskName;
    this.taskStatus = taskStatus;
    this.comments = comments;
  }
}

File: src/main/java/com/bridge/faciliwise/model/JobProcedureTask.java
package com.bridge.faciliwise.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class JobProcedureTask {
    TaskType taskType;
    String taskName;
    TaskStatus taskStatus;
    java.util.List<UserComment> comments;
}
